### PR TITLE
Comments for semver

### PR DESCRIPTION
**Pull request by DG12**

### CODE REVIEW

**Comment by ojousima** [comment]
Thank you for the contribution. 

Commenting the file server.c doesn't really belong to test/test_main.c, the comments are better to be directly in the server file. Version history is stored in git, there's no need to add manual changelog in the file header comment.

Please remove the above duplicate information, then this is ok to merge

**Comment by ojousima** [comment]
ok to test

**Comment by DG12** [comment]
As I stated in the pull request, I put the necessary comments in  test/test_main.c since semver is pulled directly from it's repository and I did not want to add comments there.

Although semver is somewhat static, It is a bad idea to directly source it from the repository.
Changes to it will be included in the ruuvi build without warning and without being tested in the ruuuvi build.

**Comment by DG12** [comment]
Very brief comments regarding changes should be in the source.
 First it is an easy reminder when first looking at the source explaining the intent. 
Have you ever tried to follow history using git? The way it is presented, with all other files changed with the same  pull request makes it nearly impossible to use. In addition the detail involving multiple request and the very minimal pull request comments like fix issue# 394 obscure the intent of the change.  

There is no down side to including comments in the source.

**Comment by DG12** [comment]
In the astyle job
You should considered adding --ignore-all-space  and  --suppress-common-lines to the diff command.
This will prevent needless failures off ovule requests.

**Comment by ojousima** [comment]
> Although semver is somewhat static, It is a bad idea to directly source it from the repository.
Changes to it will be included in the ruuvi build without warning and without being tested in the ruuuvi build.

Git submodules work by referencing a specific commit in project. That means that no matter how much much original project changes, the included source stays the same. There is a risk of project getting removed, but it's mitigated as the git submodules are pulled to local computers when fetching the project.

> There is no down side to including comments in the source.

The downside is duplicated information, it's often out of date or conflicting. Difficulties in tracking down changes in git is more about disciplined use of the tool than the tool itself, although I'll be first to admit that my own versioning is sometimes hard to follow

**Review by ojousima — APPROVED** [review]
Thanks

**Comment by ojousima** [comment]
Thank you for the contribution
